libnmstate/netapplier.py

```diff
@@ -167,6 +168,61 @@ def _apply_ifaces_state(
         raise
 
 
+def _prepare_vlan_ifaces_for_mtu_change(desired_state, current_state):
+    current_ifaces = list(six.viewitems(current_state.interfaces))
+    desired_ifaces = list(six.viewvalues(desired_state.interfaces))
+    desired_ifaces_by_name = {
+        ifstate[schema.Interface.NAME]: ifstate for ifstate in desired_ifaces
+    }
+    vlan_ifaces_states = []
+    for ifname, ifstate in current_ifaces:
+        if ifstate[schema.Interface.TYPE] != schema.InterfaceType.VLAN:
+            continue
+        vlan_dstate = desired_ifaces_by_name.get(ifname, {})
+        if (
+            vlan_dstate.get(schema.Interface.STATE)
+            == schema.InterfaceState.ABSENT
+        ):
+            continue
```

Comment:
```
current_changed_vlan_ifaces = (
(ifname, ifstate) for ifname, ifstate in current_vlan_ifaces
if desired_state.interfaces.get(ifname, {}).get(schema.Interface.STATE) == schema.InterfaceState.UP
)
```
One line here is probably too long and harder to read, so you can use a function for it: `_is_up_iface`.

Then you end up with a list comprehension like this:
```
[
_create_vlan_desired_state_from_vlan_current_state(desired_state.interfaces[ifname], ifstate, ifname)
for ifname, ifstate in current_changed_vlan_ifaces
]
```

But, on the other hand, we do this already in `State._remove_absent_interfaces`. But it looks like it is called from the verify_interface method... which smells wrong (we verify but actually update the state ://).

Reply:
I also added `_is_vlan_iface`. I tried to do it using `interfaces_to_filter=desired_vlan_names` but that would require to filter the ifaces by type too.. Is that fine? Thanks.